### PR TITLE
Use Nix's geth package & update nix lockfile

### PR DESCRIPTION
**Pull request by doubledup**
Had to switch to the unstable channel as that's got the new geth version. Running `nix flake update` after this bumped other packages like node and pnpm.

### CODE REVIEW

**Review by alistair-singh — APPROVED** [review]

**Review by vgeddes — APPROVED** [review]

File: core/packages/contracts/test/BeefyClient.t.sol [hunk 1]
```diff
@@ -112,88 +112,88 @@ contract BeefyClientTest is Test {
...
+//
+//        beefyClient.submitInitial(commitHash, bitfield);
+//
+//        cheats.roll(block.number + randaoCommitDelay);
+//
+//        cheats.difficulty(difficulty);
+//
+//        beefyClient.commitPrevRandao(commitHash);
+//
+//        BeefyClient.Commitment memory commitment = BeefyClient.Commitment(blockNumber, setId, payload);
```
Comment (yrong) [inline core/packages/contracts/test/BeefyClient.t.sol:140]: Are comments here for test CI breaking? Add some analysis here
https://snowfork.slack.com/archives/G01BGMGLAC9/p1683682087326629?thread_ts=1683621046.426719&cid=G01BGMGLAC9

File: core/packages/contracts/src/BeefyClient.sol [hunk 2]
```diff
@@ -243,8 +244,14 @@ contract BeefyClient is Ownable {
             revert TicketExpired();
         }
 
+        console.log("prevRandao before commit:");
+        console.log(ticket.prevRandao);
```
Comment (yrong) [inline core/packages/contracts/src/BeefyClient.sol:248]: would assume for debug only and better to remove for production.

File: core/packages/contracts/test/BeefyClient.t.sol (hunk 1 above)
Comment (doubledup) [reply core/packages/contracts/test/BeefyClient.t.sol:140]: Exactly, I was trying to figure out what was breaking but the Github incidents yesterday interrupted me. I'll revert the changes - best way forward is using Nix in CI to lock all of our dependencies.

**Review by yrong — APPROVED** [review]
LGTM!

**Pull request merged**